File: C++/maxDepth.cpp
#include<iostream>
#include<algorithm>
#include<cmath>

struct TreeNode {
 	int val;
 	struct TreeNode *left;
 	struct TreeNode *right;
 	TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root ==nullptr){return 0;}
        return std::max(maxDepth(root->left),maxDepth(root->right))+1;
    }
};

int main(){
    TreeNode* root = new TreeNode(1);
    root->left=new TreeNode(2);
    root->right=new TreeNode(3);
    root->left->left=new TreeNode(4);
    root->right->right=new TreeNode(5);
    Solution solution;

    int maxdepth =solution.maxDepth(root);
    std::cout<<maxdepth<<std::endl;
    return 0;
}

File: python/maxDepth.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def maxDepth(self , root: TreeNode) -> int:
        if not root:
            return 0
        return max(self.maxDepth(root.left),self.maxDepth(root.right))+1
    
root=TreeNode(1)
root.left=TreeNode(2)
root.right=TreeNode(3)
root.left.left=TreeNode(4)
root.right.right=TreeNode(5)
solution=Solution()
max_depth=solution.maxDepth(root)
print(max_depth)